addons/account/models/account_move.py

```diff
@@ -3157,26 +3157,30 @@ def add_file_data_results(file_data, invoice):
                 close_file(file_data)
                 continue
 
-            decoder = self._get_edi_decoder(file_data, new=new)
-            if decoder:
-                try:
-                    with self.env.cr.savepoint():
-                        with current_invoice._get_edi_creation() as invoice:
+            try:
+                with self.env.cr.savepoint() as savepoint:
+                    with current_invoice._get_edi_creation() as invoice:
+                        decoder = invoice._get_edi_decoder(file_data, new=new)
```

Comment:
Does it make a difference to call `_get_edi_decoder` on `self` or on `invoice`? :thinking:

Reply:
Yes, it does because the result of `_needs_auto_extract` depends also on the state of the object.
In particular, the current issue is that after sending the first pdf to ocr the invoice `extract_state` will change accordingly and we won't send further pdf to extract